iothub/device/src/Edge/CustomCertificateValidator.cs

```diff
@@ -18,59 +17,53 @@ internal class CustomCertificateValidator : ICertificateValidator
+        internal static CustomCertificateValidator Create(IList<X509Certificate2> certs, IotHubClientTransportSettings transportSettings)
         {
             var instance = new CustomCertificateValidator(certs, transportSettings);
             instance.SetupCertificateValidation();
             return instance;
         }
 
-        public Func<object, X509Certificate, X509Chain, SslPolicyErrors, bool> GetCustomCertificateValidation()
+        Func<object, X509Certificate, X509Chain, SslPolicyErrors, bool> ICertificateValidator.GetCustomCertificateValidation()
         {
-            Debug.WriteLine("CustomCertificateValidator.GetCustomCertificateValidation()");
+            if (Logging.IsEnabled)
+                Logging.Info(this, "CustomCertificateValidator.GetCustomCertificateValidation()", nameof(ICertificateValidator.GetCustomCertificateValidation));
 
             return (sender, cert, chain, sslPolicyErrors) =>
                 ValidateCertificate(_certs.First(), cert, chain, sslPolicyErrors);
         }
 
         private void SetupCertificateValidation()
         {
-            Debug.WriteLine("CustomCertificateValidator.SetupCertificateValidation()");
+            if (Logging.IsEnabled)
+                Logging.Info(this, "CustomCertificateValidator.SetupCertificateValidation()", nameof(SetupCertificateValidation));
 
             if (_transportSettings is IotHubClientAmqpSettings amqpTransportSettings)
             {
-                if (amqpTransportSettings.RemoteCertificateValidationCallback == null)
-                {
-                    amqpTransportSettings.RemoteCertificateValidationCallback =
-                        (sender, certificate, chain, sslPolicyErrors) => ValidateCertificate(_certs.First(), certificate, chain, sslPolicyErrors);
-                }
+                amqpTransportSettings.RemoteCertificateValidationCallback ??=
+                    (sender, certificate, chain, sslPolicyErrors) => ValidateCertificate(_certs.First(), certificate, chain, sslPolicyErrors);
             }
             else if (_transportSettings is IotHubClientMqttSettings mqttTransportSettings)
             {
-                if (mqttTransportSettings.RemoteCertificateValidationCallback == null)
-                {
-                    mqttTransportSettings.RemoteCertificateValidationCallback =
-                        (sender, certificate, chain, sslPolicyErrors) => ValidateCertificate(_certs.First(), certificate, chain, sslPolicyErrors);
-                }
+                mqttTransportSettings.RemoteCertificateValidationCallback ??=
+                    (sender, certificate, chain, sslPolicyErrors) => ValidateCertificate(_certs.First(), certificate, chain, sslPolicyErrors);
             }
-
-            // TODO: Tim/Abhipsa, what does this mean?
-
-            // InvokeMethodAsync is over HTTP even when transportSettings set a different protocol
```

Comment:
For reference, the cert validator for invoke methods over http is set here: https://github.com/Azure/azure-iot-sdk-csharp/blob/672b447887df5faf0018d279d4508462a761c388/iothub/device/src/IotHubModuleClient.cs#L257